core/templates/dev/head/i18n.js

```diff
@@ -45,20 +45,26 @@ oppia.constant('DEFAULT_TRANSLATIONS', {
 });
 
 oppia.controller('I18nFooter', [
-    '$rootScope', '$scope', '$translate',
-    function($rootScope, $scope, $translate) {
+    '$http', '$rootScope', '$scope', '$translate',
+    function($http, $rootScope, $scope, $translate) {
   // Changes the language of the translations.
+  var _PREFERENCES_DATA_URL = '/preferenceshandler/data';
+  var _SAVE_SITE_LANGUAGE_URL = '/save_site_language';
   $scope.supportedSiteLanguages = GLOBALS.SUPPORTED_SITE_LANGUAGES;
+  if (GLOBALS.userIsLoggedIn) {
+    $http.get(_PREFERENCES_DATA_URL).success(function(data) {
+      $translate.use(data.preferred_site_language_code);
```

Comment:
It seems a little wasteful to do a separate request on every single page the user visits. Could the backend figure this out and load it in as a global for first-load of the page?